docs/src/snap/howto/cis-hardening.md

```diff
@@ -282,6 +347,3230 @@ the sets, including the dqlite specific checks in the output.
+
+`--terminated-pod-gc-threshold=10`
+
+
+**Audit (as root):**
+
+```
+/bin/ps -ef | grep kube-controller-manager | grep -v grep
+```
+
+**Expected output:**
+
+```
+--terminated-pod-gc-threshold=12500
+```
+
+##### Control 1.3.2
+
+**Description:**
+
+Ensure that the --profiling argument is set to false
+(Automated)
+
+
+**Remediation:**
+
+Edit the Controller Manager configuration file
+/var/snap/k8s/common/args/kube-controller-manager
+on the control plane node and set the following argument.
+
+`--profiling=false`
+
+
+**Audit (as root):**
+
+```
+/bin/ps -ef | grep kube-controller-manager | grep -v grep
+```
+
+**Expected output:**
+
+```
+--profiling=false
+```
+
+##### Control 1.3.3
+
+**Description:**
+
+Ensure that the --use-service-account-credentials argument
+is set to true (Automated)
+
+
+**Remediation:**
+
+Edit the Controller Manager configuration file
+/var/snap/k8s/common/args/kube-controller-manager
+on the control plane node to set the following argument.
+
+`--use-service-account-credentials=true`
+
+
+**Audit (as root):**
+
+```
+/bin/ps -ef | grep kube-controller-manager | grep -v grep
+```
+
+**Expected output:**
+
+```
+--user-service-account-credentials=true
+```
+
+##### Control 1.3.4
+
+**Description:**
+
+Ensure that the --service-account-private-key-file argument
+is set as appropriate (Automated)
+
+
+**Remediation:**
+
+Edit the Controller Manager configuration file
+/var/snap/k8s/common/args/kube-controller-manager
+on the control plane node and set the --service-account-
+private-key-file parameter
+to the private key file for service accounts.
+
+`--service-account-private-key-file=<filename>`
+
+
+**Audit (as root):**
+
+```
+/bin/ps -ef | grep kube-controller-manager | grep -v grep
+```
+
+**Expected output:**
+
+```
+--service-account-private-key-
+file=/etc/kubernetes/pki/serviceaccount.key
+```
+
+##### Control 1.3.5
+
+**Description:**
+
+Ensure that the --root-ca-file argument is set as
+appropriate (Automated)
+
+
+**Remediation:**
+
+Edit the Controller Manager configuration file
+/var/snap/k8s/common/args/kube-controller-manager
+on the control plane node and set the --root-ca-file
+parameter to the certificate bundle file.
+
+`--root-ca-file=<path/to/file>`
+
+
+**Audit (as root):**
+
+```
+/bin/ps -ef | grep kube-controller-manager | grep -v grep
+```
+
+**Expected output:**
+
+```
+--root-ca-file=/etc/kubernetes/pki/ca.crt
+```
+
+##### Control 1.3.6
+
+**Description:**
+
+Ensure that the RotateKubeletServerCertificate argument is
+set to true (Automated)
+
+
+**Remediation:**
+
+Edit the Controller Manager configuration file
+/var/snap/k8s/common/args/kube-controller-manager
+on the control plane node and set the --feature-gates
+parameter to include RotateKubeletServerCertificate=true.
+
+`--feature-gates=RotateKubeletServerCertificate=true`
+
+
+**Audit (as root):**
+
+```
+/bin/ps -ef | grep kube-controller-manager | grep -v grep
+```
+
+**Expected output:**
+
+```
+--feature-gates is not set
```

Comment:
expected output does not match remediation

Reply:
Fixed, see https://github.com/canonical/k8s-snap/pull/802/files#diff-da8b48161c214d6854a37a2ef8a664491413a6b43793ba6e9e6319a49da719ecR2063